pandas/tests/indexes/interval/test_interval.py

```diff
@@ -420,8 +413,26 @@ def test_get_loc_value(self):
         assert idx.get_loc(3) == 1
         pytest.raises(KeyError, idx.get_loc, 3.5)
 
-        idx = IntervalIndex.from_arrays([0, 2], [1, 3])
-        pytest.raises(KeyError, idx.get_loc, 1.5)
+    # TODO: add overlapping/non-monotonic/duplicate index tests
+    @pytest.mark.parametrize("scalar", [-0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5])
+    def test_get_loc_scalar(self, closed, scalar):
```

Comment:
can you put errors in a separate test (test_get_loc_scalar_error)

Reply:
Yes, but imo doing so would be less concise add code bloat.  Could certainly do it, but want to make the downsides known first.

By separating them, you can't really parametrize over both `scalar` and `closed`, since testing endpoints will raise for some values of `closed` but pass for others, so one of them will need to be written inside a for loop of some kind.  It'd also add a bit of repetitiveness, since you'd need to encode passing keys in the passing test, and failing keys in the errors tests, whereas the current version only encodes passing with the missing values presumed to be failing.